databaseAnnotator/src/main/java/org/cbioportal/database/annotator/AnnotateRecordsProcessor.java

```diff
@@ -0,0 +1,165 @@
+import org.cbioportal.annotator.Annotator;
+import org.cbioportal.database.annotator.model.*;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.client.*;
+
+/**
+ *
+ * @author heinsz
+ */
+public class AnnotateRecordsProcessor implements ItemProcessor<MutationEvent, MutationEvent>{
+    @Value("#{jobParameters[isoform]}")
+    private String isoform;
+
+    @Autowired
+    Annotator annotator;
+
+    @Autowired
+    SQLQueryFactory databaseAnnotatorQueryFactory;
+
+    public final static int NA_INT = -1;
+
+    private final Logger log = Logger.getLogger(AnnotateRecordsProcessor.class);
+
+    @Override
+    public MutationEvent process(MutationEvent i) throws Exception {
+        // first make a MutationRecord that we can send to Genome Nexus to annotate
+        Map<String, String> mafLine = new HashMap<>();
+        String chromosome;
+        if (i.getCHR().equals("23")) {
+                chromosome = "X";
+        }
+        else if (i.getCHR().equals("24")) {
+                chromosome = "Y";
+        }
+        else {
+                chromosome = i.getCHR();
+        }
+        mafLine.put("Chromosome", chromosome);
+        mafLine.put("Start_Position", String.valueOf(i.getSTART_POSITION()));
+        mafLine.put("End_Position", String.valueOf(i.getEND_POSITION()));
+        mafLine.put("Reference_Allele", i.getREFERENCE_ALLELE());
+        mafLine.put("Tumor_Seq_Allele2", i.getTUMOR_SEQ_ALLELE());
+        MutationRecord record = annotator.createRecord(mafLine);
+        AnnotatedRecord annotatedRecord;
+        try {
+                annotatedRecord = annotator.annotateRecord(record, true, isoform, true);
+        }
+        catch (HttpServerErrorException | ResourceAccessException e) {
+           log.error("Failed to annotate record - errors accessing Genome Nexus");
+           return i;
+        }
+
+        // If the mutation is a 5' or 3' flank mutation, or any other UTR, it shouldn't be in the database as these are supposed to be filtered.. Remove it.
+		if (annotatedRecord.getVariant_Classification().equals("5'Flank") && annotatedRecord.getHugo_Symbol().equals("TERT")) {
+			i.setPROTEIN_CHANGE("Promoter");
+		}
+        else if (annotatedRecord.getVariant_Classification().equals("5'Flank") || 
+                annotatedRecord.getVariant_Classification().equals("3'Flank") || 
+                annotatedRecord.getVariant_Classification().equals("IGR") || 
+                annotatedRecord.getVariant_Classification().equals("Intron") || 
+                annotatedRecord.getVariant_Classification().equals("RNA") || 
+                annotatedRecord.getVariant_Classification().equals("5'UTR") ||
+                annotatedRecord.getVariant_Classification().equals("3'UTR")) {
+            deleteMutation(i);
+            i.setPROTEIN_CHANGE("MUTATED");
```

Comment:
return null here instead of setting protein change to MUTATED since we're dropping these types of records from the mutation/mutation_event tables anyway